### DISCUSSION TITLE
Refer to other workflows and share artifacts?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#62613] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello!
I am writing a VSCode Extension. I need to write a GitHub Action that packages the extension first (as .vsix), uploads that as an artifact, and then calls a publish-msmarketplace and publish-openvsix to publish the extension to both MS Marketplace and OpenVSIX. However, this must only happen when a tag is pushed. When a commit is pushed, I just want to upload the artifact.

The code seems fine to me but I am getting [this error](https://github.com/aitor-gomila/vscode-language-pack-ca/actions/runs/5751655342) from GH Actions. Apparently it's looking for an "actions.yml" file tho I couldn't find any docs on it. [Source code](https://github.com/aitor-gomila/vscode-language-pack-ca/blob/main/.github/workflows/package.yml).

**Comment by mickeygousset** [comment, marked as answer]
You aren't calling your reusable workflows correctly on lines 39-42.

Each reusable workflow is considered to be its own job step.  So you need to create a new job for each reusable workflow you want to call.

Example of calling a reusable workflow:

https://github.com/mickeygoussetorg/github-actions-presentation/blob/main/.github/workflows/mywebapp-publish-release-with-reusable-workflow.yml

Docs: https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

**Reply by ghost** [reply]
Thanks! How quick!

**Reply by ghost** [reply]
Additionally, I need to run these deploy workflows when the current checkout is a tag. I am doing that with a command which errors with exit code 128 when it is a tag. Is there a way to do that?

**Reply by mickeygousset** [reply]
One of the events you can trigger a workflow on is the creation of a tag.
